### PR TITLE
Skip kube-up related GCE tests when running e2e tests on kops

### PR DESCRIPTION
**Pull request by upodroid**
#### What this PR does / why we need it:
I'm migrating away from kube-up clusters to kops clusters for k/k e2e testing. Some of our e2e tests are written to test functionality that is only present in kube-up so those tests now only run on kube-up clusters. 

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/test-infra/pull/30686

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by BenTheElder** [comment]
Not thrilled about adding a new framework var and doubly not thrilled about needing to set this flag all over the place. Kube-up CI has been running for ages in many places and shouldn't require a breaking change.

thinking ...

**Comment by aojea** [comment]
/hold

we don't want to make the e2e test framework smart, and the goal is to remove all these cloud provider skips from the framework not add more ... the jobs should use the regex to skip tests, Patrick also had a kep to improve this mechanism using ginkgo tags

**Comment by aojea** [comment]
@upodroid you definitely have to write a  KEP under sig-testing if you plan to touch on kubernetes code base and making impactful changes on jobs and behaviors, so the SIGs can evaluate and be aware of the changes https://github.com/kubernetes/enhancements/tree/master/keps/sig-testing

**Comment by pohly** [comment]
I agree, `TestContext.Tooling` looks like a candidate for removal. My PR for labeling is https://github.com/kubernetes/kubernetes/pull/112894 (will get back to it soonish) and the kind-of-related KEP is https://github.com/kubernetes/enhancements/tree/master/keps/sig-testing/3041-node-conformance-and-features (might need a new owner).

**Comment by upodroid** [comment]
/retest

As we discussed yesterday, I'm writing a doc that will form the KEP for this project.

In the meanwhile, can we remove the tests in `test/e2e/network/firewall.go` file? It tests gcp infra that is created by kube-up scripts, has nothing to do with Kubernetes functionality or the gcp cloud provider. If we can do that, then I don't have to introduce extra logic that uses the `tooling` flag.

File: test/e2e/framework/test_context.go [hunk 1]
```diff
@@ -398,7 +398,7 @@ func RegisterClusterFlags(flags *flag.FlagSet) {
 	// If this becomes true as well, they should be refactored into RegisterCommonFlags.
 	flags.BoolVar(&TestContext.PrepullImages, "prepull-images", false, "If true, prepull images so image pull failures do not cause test failures.")
 	flags.StringVar(&TestContext.Provider, "provider", "", "The name of the Kubernetes provider (gce, gke, local, skeleton (the fallback if not set), etc.)")
-	flags.StringVar(&TestContext.Tooling, "tooling", "", "The tooling in use (kops, gke, etc.)")
+	flags.StringVar(&TestContext.Tooling, "tooling", "kube-up", "The tooling used to create the cluster (kops, gke, kube-up).")
```
Comment (aojea) [inline test/e2e/framework/test_context.go:401]: we should not set this default

Comment (upodroid) [reply test/e2e/framework/test_context.go:401]: This PR is being closed out. I needed to check something in the kubetest2 test.

/close

**Pull request closed without merging**